lib/utilization/azurefunction-info.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const logger = require('../logger.js').child({ component: 'azurefunction-info' })
+const NAMES = require('../metrics/names.js')
+let results = null
+
+module.exports = fetchAzureFunctionInfo
+module.exports.clearCache = function clearAzureFunctionCache() {
+  results = null
+}
+
+function fetchAzureFunctionInfo(agent, callback) {
+  if (!agent.config.utilization || !agent.config.utilization.detect_azurefunction) {
+    return setImmediate(callback, null, null)
+  }
+
+  if (results) {
+    return setImmediate(callback, null, results)
+  }
+
+  // Detect if Azure Function.
+  // TODO: How do we access Azure Function invocation context?
```

Comment:
The agent would be attached to the Azure Function app code in order to execute, so we wouldn't have to worry about the environment except to simulate it during testing. To get the utilization info, these environment variables would be made available as properties of `process.env`.